docker/utils/utils.py

```diff
@@ -100,7 +101,14 @@ def tar(path, exclude=None, dockerfile=None, fileobj=None):
     exclude = exclude or []
 
     for path in sorted(exclude_paths(root, exclude, dockerfile=dockerfile)):
-        t.add(os.path.join(root, path), arcname=path, recursive=False)
+        i = t.gettarinfo(os.path.join(root, path), arcname=path)
+        if sys.platform == 'win32':
+            i.mode = i.mode & 0o755 | 0o111
+        if i.isreg():
+            with open(os.path.join(root, path), 'rb') as f:
+                t.addfile(i, f)
```

Comment:
I did play around with the `tarfile` module in the Python interpreter on Linux and was just getting zeroed files in my tarball without it. However `docker.Client.build()` seems to work fine without it. I guess we can drop it then.